packages/core/src/app/address/googleAutocomplete/GoogleAutocomplete.test.tsx

```diff
@@ -0,0 +1,33 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+
+import GoogleAutocomplete from './GoogleAutocomplete';
+
+describe('GoogleAutocomplete Component', () => {
+    // @todo: add more tests when we can successfully mock "GoogleAutocompleteService"
+    // At the moment, jest.mock('...') doesn't seem to do the trick.
+
+    it('renders input with initial value', () => {
+        expect(
+            render(
+                <GoogleAutocomplete
+                    apiKey="bar"
+                    initialValue="fo"
+                    inputProps={{ placeholder: 'NO PO BOX' }}
+                />,
+            ),
+        ).toMatchSnapshot();
+    });
+
+    it('renders input with max length', () => {
+        expect(
+            render(
+                <GoogleAutocomplete
+                    apiKey="bar"
+                    initialValue="fo"
+                    inputProps={{ maxLength: "50" }}
+                />,
+            ),
+        ).toMatchSnapshot();
```

Comment:
Can we test `GoogleAutocomplete` in the way the user would use it?

Reply:
Yes I will say probably need to avoid snapshot testing.

Reply:
@bc-peng @animesh1987 I tried mocking google maps service following some suggestions [here](https://github.com/hibiken/react-places-autocomplete/issues/189) and [here](https://stackoverflow.com/questions/58439971/how-can-i-test-google-maps-geocoder) but those only give a way to check if the service has been called `n` number of times and not give the actual results like we get from the maps service. Hence, I cannot add more tests to check if the services work fine and give correct results. However, I was able to add tests to check the scenario for max-length and placeholder texts using RTL and jest and avoid snapshot testing.